### DISCUSSION TITLE
Get-PnPSiteTemplate to make template of list with data, folders and attachments.

### DISCUSSION THREAD

**Discussion by alexislagodka** [pnp/powershell#517] [Q&A]
I want to make a template of list include items, folders and attachments.
I used : 

```
Connect-PnPOnline -Url $siteurl -Interactive
Get-PnPSiteTemplate -Configuration configuration.json -Out template.xml
```

With the configuration file below :

```
{
    "$schema":"http://json-schema.org/draft-07/schema",
    "persistAssetFiles": true,
    "handlers": [
      "Lists",
      "WebSettings",
      "Pages"
    ],
    "lists": {
      "lists": [
        {
          "title": "test 1",
          "includeItems": true,
          "query": {
            "includeAttachments": true
          }
        }
      ]
    },
    "pages": {
      "includeAllClientSidePages": true
    }
  }
```
But the template return items with no data in rows, no folders ,but it works for attachments.

I tried another approach with these commands :

```
Connect-PnPOnline -Url $siteurl -Interactive
Get-PnPSiteTemplate -Handlers Lists -ListsToExtract 'test 1' -Out 'template.xml'
Add-PnPListFoldersToSiteTemplate -Path 'template.xml' -List 'test 1' -Recursive -IncludeSecurity
Add-PnPDataRowsToSiteTemplate -path $pathListTemplate -List 'test 1'
```
This solution works for items, folders but not for items in folders and not for attachments.

So, anyone have an idea to do that ?
Or it's not possible to do that with the pnp provisioning engine ?

**Comment by kasperbolarsen** [comment]
As far as I know you must use the pnp format in order for it to include content such as list items, at least that works for me
